sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -403,19 +405,25 @@ ProgramManager::getBuiltPIProgram(OSModuleHandle M, const context &Context,
     return BuiltProgram.release();
   };
 
-  return getOrBuild<PiProgramT, compile_program_error>(Cache, KSId, AcquireF,
-                                                       GetF, BuildF);
+  using KeyT = KernelProgramCache::ProgramCacheKeyT;
+  SerializedObj SpecConsts;
+  if (Prg)
+    Prg->stableSerializeSpecConstRegistry(SpecConsts);
+
+  return getOrBuild<PiProgramT, compile_program_error>(
+      Cache, KeyT(std::move(SpecConsts), KSId), AcquireF, GetF, BuildF);
```

Comment:
Should these programs (i.e. with specialization consts) be cached at all?

Reply:
Yes. Why it should not?